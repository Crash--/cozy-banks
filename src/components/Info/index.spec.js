import React from 'react'
import { Infos } from '.'
import { shallow } from 'enzyme'

describe('Info', () => {
  it('should render its children', () => {
    expect(shallow(<Infos>Children</Infos>).getElement()).toMatchSnapshot()
  })

  it('should render with a title', () => {
    expect(
      shallow(<Infos title="title">Children</Infos>).getElement()
    ).toMatchSnapshot()
  })

  it('should render with an icon', () => {
    expect(
      shallow(<Infos icon="openwith">Children</Infos>).getElement()
    ).toMatchSnapshot()
  })

  it('should render with the right variant', () => {
    expect(
      shallow(<Infos variant="error">Children</Infos>).getElement()
    ).toMatchSnapshot()
  })
})
