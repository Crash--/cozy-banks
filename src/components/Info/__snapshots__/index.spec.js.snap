// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should render its children 1`] = `
<div
  className="u-p-1 infos--normal"
>
  <div
    className="infos--container"
  >
    <div
      className="infos--content"
    >
      Children
    </div>
  </div>
</div>
`;

exports[`Info should render with a title 1`] = `
<div
  className="u-p-1 infos--normal"
>
  <div
    className="infos--container"
  >
    <div
      className="infos--content"
    >
      <SubTitle
        className="infos--title"
        ellipsis={false}
        tag="div"
      >
        title
      </SubTitle>
      Children
    </div>
  </div>
</div>
`;

exports[`Info should render with an icon 1`] = `
<div
  className="u-p-1 infos--normal"
>
  <div
    className="infos--container"
  >
    <Icon
      className="infos--icon"
      icon="openwith"
      spin={false}
    />
    <div
      className="infos--content u-pl-half"
    >
      Children
    </div>
  </div>
</div>
`;

exports[`Info should render with the right variant 1`] = `
<div
  className="u-p-1 infos--error"
>
  <div
    className="infos--container"
  >
    <div
      className="infos--content"
    >
      Children
    </div>
  </div>
</div>
`;
